ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -333,6 +333,60 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       }
     }
   }
+  
+  func deleteSelection() -> Bool {
+    if let selected = textDocumentProxy.selectedText, selected.count > 0 {
```

Comment:
This guard ensures we only try to delete selected text only if there actually _is_ selected text - a selection exists with length > 0.  If it doesn't exist, we exit and indicate that no deletion happened.